### PR TITLE
Bug 1540916 - Use correct HPA scaleRef apiVersion

### PR DESCRIPTION
**Pull request by spadgett**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1540916

/assign @jwforres 
/cc @benjaminapetersen

### CODE REVIEW

**Comment by spadgett** [comment]
`apps.openshift.io` has a `deploymentconfigs/scale` subresource. (It used to be in the extensions API group.) I'm checking all of the others to be sure.

openshift web console 2018-02-01 08-29-55

**Comment by spadgett** [comment]
We should change our handling of the scale subresource in DeploymentsService. Related to https://github.com/openshift/origin-web-console/issues/2723#issuecomment-362097290

**Comment by spadgett** [comment]
Updated DeploymentsService, which makes scaling stateful sets on the overview work (#2723).

**Comment by spadgett** [comment]
Note that by default project admins don't have authority to update the statefulsets/scale subresource, but it works for cluster admins.

File: app/views/browse/_replica-set-details.html [hunk 1]
```diff
@@ -198,7 +198,7 @@ <h3>Autoscaling</h3>
          ng-href="project/{{projectName}}/edit/autoscaler?kind=ReplicationController&name={{replicaSet.metadata.name}}"
          role="button">Add Autoscaler</a>
       <a ng-if="replicaSet.kind === 'ReplicationController' && deploymentConfigName"
-         ng-href="project/{{projectName}}/edit/autoscaler?kind=DeploymentConfig&name={{deploymentConfigName}}"
+         ng-href="project/{{projectName}}/edit/autoscaler?kind=DeploymentConfig&name={{deploymentConfigName}}&group=apps.openshift.io"
```
Comment (benjaminapetersen) [inline app/views/browse/_replica-set-details.html]: Wondering if we should have things like `&group=apps.openshift.io` in our templates, or if that makes it easy to miss things later.  Could we potentially keep this in a variable in the controller near the `preferredVersion` stuff?

Just trying to think if we can avoid potential mismatches in the future, though I'm sure group changes will be fairly minimal (this is more of a question than a request for change).

File: app/views/browse/stateful-set.html [hunk 2]
```diff
@@ -63,14 +63,16 @@ <h1>
               <div class="row" style="max-width: 650px;">
                 <div class="col-sm-4 col-sm-push-8 browse-deployment-donut">
                   <!--
```
Comment (benjaminapetersen) [inline app/views/browse/stateful-set.html]: Assuming `autoscalers` would be part of issue #2723 work?

**Comment by spadgett** [comment]
I'm going to break this down into smaller fixes.

/close

**Pull request closed without merging**